cf-agent/verify_exec.c

```diff
@@ -294,18 +295,18 @@ static ActionResult RepairExec(EvalContext *ctx, const Attributes *a,
     if (a->transaction.background)
     {
 #ifdef __MINGW32__
-        outsourced = true;
+        do_work_here = true;
 #else
         Log(LOG_LEVEL_VERBOSE, "Backgrounding job '%s'", cmdline);
-        outsourced = fork();
+        do_work_here = (fork() == 0); // true for child, false for parent
 #endif
     }
     else
     {
-        outsourced = false;
+        do_work_here = false;
```

Comment:
This should be `do_work_here = true`, right? I know it makes no difference now, but the condition below should be changed to not check `a->transaction.background` again. Let's just use this one `if` to decide where to do the work.

Reply:
Ye, but then it becomes a little dangerous/unclear whether I should change behavior or not. Because for windows it is different, and `do_work_here` is checked further down. Tried to address this in another commit.